subsys/settings/src/settings_fcb.c

```diff
@@ -359,6 +359,14 @@ static int settings_fcb_save(struct settings_store *cs, const char *name,
 		return -EINVAL;
 	}
 
+	/*
+	 * Equal signs is used internally as separator between name and value. Reject in order to
+	 * not corrupt the saved value.
+	 */
+	if (strchr(name, '=')) {
```

Comment:
> However, this will break applications using the NVS settings backend and equal signs in their names.

See the point.

> How about add a Kconfig to allow disabling this check? Could keep it for a while (not active per default), mark it as deprecated and then remove after a while?

That sounds to me like not a good practice.

I was thinking about possible alternative which will allow to use '=' even with the fcb backend.

I looked through the code and it looks like it would be possible to convert '=' to something else, a nonprinting character.
Looks like small patchces would be needed only to settings_line_write(), settings_line_name_read().

Reply:
> I was thinking about possible alternative which will allow to use '=' even with the fcb backend.

That would be the kind of not-that-simply solution which I originally wanted to avoid. ;)

> I looked through the code and it looks like it would be possible to convert '=' to something else, a nonprinting character. Looks like small patchces would be needed only to settings_line_write(), settings_line_name_read().

While (very?) uncommon, any character could have been used in the past as part of the name. Well, except for '\0'.

How about changing the code to write '\0' as limiter (instead of the equal sign) and adding some migration code in the read parts of the settings fcb subsystem to deal with existing equal sign delimiters?

Reply:
@nvlsianpu After thinking for about this for some other days, I still think that accepting this backend specific limitation (flat out denying having any equal sign chars in the name) is the best way to go. Better than using any non-printing characters for the reasons outlined above.

Reply:
Just realized: There is SETTINGS_FCB_VERS - it might be possible to bump this version and then to use it as indication of which limiter (old: '=', new: '\n') to use. I think this might be the best solution so far, but I am not sure if and when I will have to time to implement this.

Reply:
> @nvlsianpu After thinking for about this for some other days, I still think that accepting this backend specific limitation (flat out denying having any equal sign chars in the name) is the best, cleanest way to move forward. Better than using any non-printing characters for the reasons outlined above.

Backends should be able to store the same information, regardless what it is. Differentiating between backends start problems, as people will have to select backend depending on what they want to store, and will not be able to switch backeds. This is not good for subsystem.

Reply:
> Just realized: There is SETTINGS_FCB_VERS - it might be possible to bump this version and then to use it as indication of which limiter (old: '=', new: '\n') to use. I think this might be the best solution so far, but I am not sure if and when I will have to time to implement this.

Good idea. Bum that might be used for overcome the issue. Need to be sure to be backward compatible.

Reply:
@rettichschnidi Are you able to continue work on this PR?

Reply:
@de-nordic As the scope of this PR has grown and my employer is not using FCB as settings backend, this issue became low on my priority list. Realistically speaking, I will not be working on it for a long time to come, if at all. If anyone is willing to take over I'd be very happy to drop this PR.